### ISSUE TITLE
Support hierarchical delimiters and qualifiers.

### ISSUE THREAD

**Issue by prb** [FasterXML/jackson-dataformat-csv#21]
The goal would be to transform nested delimiters into nested structures; sample input:

```
a,b;c;d,e;f
```

Sample output:

```
[
  [a],
  [b,c,d],
  [e,f]
]
```

(Brought over here from a discussion in FasterXML/jackson#2.)

**Comment by mlvn23** [comment]
Maybe implement this feature like this: https://github.com/Keyang/node-csvtojson#empowered-json-parser. I think this allows arbitrary nesting of data in json while still easily editable in spreadsheets.

**Comment by cowtowncoder** [comment]
Yes, use of naming convention can help. FWIW, use of `@JsonUnwrapped` already works, so while not as convenient, this is already partially doable. But more fluent support is sort of planned; it just requires integration with introspection (traversal of POJO properties) to produce logical names, and then handling nesting.
And by planned I just mean "thought about regarding feasibility" :)
